admin_guide/node_problem_detector.adoc

```diff
@@ -0,0 +1,426 @@
+** corrupted file system
+** unresponsive runtime daemon
+
+* infrastructure daemon issues: 
+** NTP service outage
+
+You can use the events and conditions produced by the Node Problem Detector 
+as a signal that a recovery action needs to be performed on the node.
+
+{product-title} supports the following problem daemons. Installation of the daemons is beyond the scope of this document.  
+
+|===
+| Problem Daemon | NodeCondition | Description
+
+| link:https://github.com/kubernetes/node-problem-detector/blob/master/config/kernel-monitor.json[KernelMonitor]
+| KernelDeadlock
+| A system log that monitors kernel logs and reports problem according to predefined rules.
+
+| link:https://github.com/abrt/abrt/wiki[AbrtAdaptor]
+| None
+| Monitor ABRT log messages. ABRT (Automatic Bug Report Tool) is a health monitoring daemon able 
+to catch kernel problems as well as application crashes on the host.
+
+| link:https://github.com/kubernetes/node-problem-detector/blob/master/config/custom-plugin-monitor.json[CustomPluginMonitor]
+| On-demand (based on user configuration)
+| A custom plugin monitor that checks various node problems with xref:admin-guide-node-problem-detector-monitor[custom, user-defined scripts].
+
+|===
+
+//https://kubernetes.io/docs/tasks/debug-application-cluster/monitor-node-health/#node-problem-detector
+The node problem detector supports file-based kernel logging. However, it is easy to extend it to xref:admin-guide-node-problem-detector-log[support other log formats].
+
+== Using the Node Problem Detector
+
+Verify that the node problem detector is running:
+
+[source,bash]
+----
+# oc get pods -n openshift-node-problem-detector
+# oc logs -n openshift-node-problem-detector <pod_name>
+----
+
+For example:
+
+[source,bash]
+----
+# oc logs -n openshift-node-problem-detector node-problem-detector-c6kng
+I0416 23:22:00.641354       1 log_monitor.go:63] Finish parsing log monitor config file: {WatcherConfig:{Plugin:journald PluginConfig:map[source:kernel] LogPath:/host/log/journal Lookback:5m} BufferSize:10 Source:kernel-monitor DefaultConditions:[{Type:KernelDeadlock Status:false Transition:0001-01-01 00:00:00 +0000 UTC Reason:KernelHasNoDeadlock Message:kernel has no deadlock}]
+----
+
+To use the node problem detector, use the `oc get` command with a resource:
+
+[source,bash]
+----
+# oc get node registry-router-1 -o yaml | grep -B5 KernelDeadlock
+
+ - lastHeartbeatTime: 2018-04-17T08:01:44Z
+    lastTransitionTime: 2018-04-17T08:01:43Z
+    message: task docker:1234 blocked for more than 700 seconds.
+    reason: DockerHung
+    status: "True"
+----
+
+[[admin-guide-node-problem-detector-overwrite]]
+== Configuring the Node Problem Detector
+
+You can configure the Node Problem Detector to add or remove problem conditions and events.
+
+. Edit the Node Problem Detector configuration map with a text editor.
++
+[source,bash]
+----
+oc edit ds -n openshift-node-problem-detector configmap node-problem-detector
```

Comment:
Sorry if I told you wrong before. This one should not have `ds` in it. This command is to edit the `configmap` (which comes later in the command).